src/value/borrowed.rs

```diff
@@ -59,8 +67,54 @@ impl<'val> BorrowedSymbolToken<'val> {
             source,
         }
     }
+
+    /// Decorates the [`BorrowedSymbolToken`] with text.
+    pub fn with_text(mut self, text: &'val str) -> Self {
+        self.text = Some(text);
+        self
+    }
+
+    /// Decorates the [`BorrowedSymbolToken`] with a local ID.
+    pub fn with_local_sid(mut self, local_sid: SymbolId) -> Self {
+        self.local_sid = Some(local_sid);
+        self
+    }
+
+    /// Decorates the [`BorrowedSymbolToken`] with an [`BorrowedImportSource`].
+    pub fn with_source(mut self, table: &'val str, sid: SymbolId) -> Self {
+        self.source = Some(BorrowedImportSource::new(table, sid));
+        self
+    }
+}
+
+/// Constructs an [`BorrowedSymbolToken`] with unknown text and a local ID.
+/// A common case for binary parsing (though technically relevant in text).
+#[inline]
+pub fn borrowed_local_sid_token<'val>(local_sid: SymbolId) -> BorrowedSymbolToken<'val> {
+    BorrowedSymbolToken::new(None, Some(local_sid), None)
+}
+
+/// Constructs an [`BorrowedSymbolToken`] with just text.
+/// A common case for text and synthesizing tokens.
+#[inline]
+pub fn borrowed_text_token(text: &str) -> BorrowedSymbolToken {
+    BorrowedSymbolToken::new(Some(text), None, None)
+}
+
```

Comment:
I would not call this `borrowed_` I would just call this `text_token` and re-alias in import.  Also codecov is saying these were not tested which surprises me.